src/commands/cmd_zset.cc

```diff
@@ -266,6 +269,77 @@ class CommandZPopMax : public CommandZPop {
   CommandZPopMax() : CommandZPop(false) {}
 };
 
+class CommandZMPop : public Commander {
+ public:
+  CommandZMPop() = default;
+
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 1);
+    numkeys_ = GET_OR_RET(parser.TakeInt<int>(NumericRange<int>{1, std::numeric_limits<int>::max()}));
+    for (int i = 0; i < numkeys_; ++i) {
+      keys_.emplace_back(GET_OR_RET(parser.TakeStr()));
+    }
+    bool has_min_flag = false;
+
+    while (parser.Good()) {
+      if (parser.EatEqICase("min")) {
+        min_ = true;
+        has_min_flag = true;
+      } else if (parser.EatEqICase("max")) {
+        min_ = false;
+        has_min_flag = true;
+      } else if (parser.EatEqICase("count")) {
+        count_ = GET_OR_RET(parser.TakeInt<int>(NumericRange<int>{1, std::numeric_limits<int>::max()}));
+      } else {
+        return parser.InvalidSyntax();
+      }
+    }
+    if (!has_min_flag) {
+      return parser.InvalidSyntax();
+    }
+    return Commander::Parse(args);
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::ZSet zset_db(svr->storage, conn->GetNamespace());
+    for (auto &user_key : keys_) {
+      std::vector<MemberScore> member_scores;
+      auto s = zset_db.Pop(user_key, count_, min_, &member_scores);
+      if (!s.ok()) {
+        return {Status::RedisExecErr, s.ToString()};
+      }
+      if (member_scores.empty()) {
+        continue;
+      }
+
+      output->append(redis::MultiLen(2));
+      output->append(redis::BulkString(user_key));
+      output->append(redis::MultiLen(member_scores.size() * 2));
+      for (const auto &ms : member_scores) {
+        output->append(redis::BulkString(ms.member));
+        output->append(redis::BulkString(util::Float2String(ms.score)));
+      }
+      return Status::OK();
+    }
+    *output = redis::NilString();
+    return Status::OK();
+  }
+
+  static CommandKeyRange Range(const std::vector<std::string> &args) {
+    int num_key = *ParseInt<int>(args[1], 10);
+    if (static_cast<int>(args.size()) > num_key + 3) {
+      return {2, 1 + num_key, 1};
+    }
+    return {2, -2, 1};
```

Comment:
```suggestion
    return {2, 1 + num_key, 1};
```

Sorry, I notice that the `if` is not necessary actually. :rofl:

Reply:
Fixed.